### PR TITLE
#446 Check for session first before logging out in fastLogout method of DrupalAuthenticationManager

### PR DESCRIPTION
**Pull request by m4olivei**
Attempt at resolving #446.

### CODE REVIEW

**Comment by m4olivei** [comment]
This patch works for me. I think it's a Selenium specific issue, where the Selenium Mink Driver is assuming that the session is already started when you call `reset` here: https://github.com/minkphp/MinkSelenium2Driver/blob/v1.3.1/src/Selenium2Driver.php#L359. When it's not, 💥 .

**Comment by m4olivei** [comment]
Incidentally, I see there is some awesome test coverage in this package. I wonder if it would be possible to have some test cases with Mink Selenium2 Driver and a browser like Chrome?

**Comment by jhedstrom** [comment]
> Incidentally, I see there is some awesome test coverage in this package. I wonder if it would be possible to have some test cases with Mink Selenium2 Driver and a browser like Chrome?

That would be great! Happy to review PRs that update `.travis.yml` to add some coverage there (and perhaps a new suite or profile in `behat.yml.dist` to specifically target selenium...

File: src/Drupal/DrupalExtension/Manager/DrupalAuthenticationManager.php [hunk 1]
```diff
@@ -130,7 +130,10 @@ public function loggedIn()
      */
     public function fastLogout()
     {
-        $this->getSession()->reset();
-        $this->userManager->setCurrentUser(false);
+        $session = $this->getSession();
+        if ($session->isStarted()) {
+            $session->reset();
+            $this->userManager->setCurrentUser(false);
```
Comment (jhedstrom) [inline src/Drupal/DrupalExtension/Manager/DrupalAuthenticationManager.php]: I wonder if we should unset the user outside the conditional?

Comment (m4olivei) [reply src/Drupal/DrupalExtension/Manager/DrupalAuthenticationManager.php]: That makes sense, will update.

**Comment by m4olivei** [comment]
> That would be great! Happy to review PRs that update .travis.yml to add some coverage there (and perhaps a new suite or profile in behat.yml.dist to specifically target selenium...

I took an honest look at it (being new to Travis), and it seems like it would be tricky to get all the dependencies right to be able to run chrome in a window manager and hook selenium up to it. I've seen it done in CircleCI before. Although it would be nice, I'm going to slowly climb out of this rabbit hole for now :).

**Comment by jhedstrom** [comment]
Thanks!

**Comment by jhedstrom** [comment]
@m4olivei I found [this Travis setup for the mink selenium driver](https://github.com/minkphp/MinkSelenium2Driver/blob/master/.travis.yml), it might help you out if you get the chance :)

**Pull request merged**